src/write.rs

```diff
@@ -0,0 +1,96 @@
+//! Customize serialization behaviour
+
+use std::{borrow::Cow, fs::File, io::BufWriter, path::Path};
+
+use plist::XmlWriteOptions;
+
+use crate::Error;
+
+/// Options that can be set when writing the UFO to disk.
+#[derive(Debug, Clone)]
+pub struct WriteOptions {
+    // for annoying reasons we store three different representations.
+    pub(crate) indent_str: Cow<'static, str>,
+    xml_opts: XmlWriteOptions,
+    pub(crate) whitespace_char: u8,
+    pub(crate) whitespace_count: usize,
+}
+
+impl Default for WriteOptions {
+    fn default() -> Self {
+        WriteOptions {
+            indent_str: "\t".into(),
+            xml_opts: Default::default(),
+            whitespace_char: b'\t',
+            whitespace_count: 1,
+        }
+    }
+}
+
+impl WriteOptions {
+    /// Create a new set of options.
+    pub fn new() -> WriteOptions {
+        WriteOptions::default()
+    }
+
+    /// Builder-style method to customize the whitespace.
```

Comment:
The WriteOptions are instantiated like this?:

```rust
// single tab
let wo = WriteOptions::new().whitespace("\t");
```

```rust
// two spaces
let wo = WriteOptions::new().whitespace("  ");
```

Reply:
Yea, I don't love this API, but the idea is when there are more options (like the ' vs " char for the declaration) then those are also set here, and you can call any subset of the builder methods.

Reply:
Sgtm.  Ty!

Reply:
The other side of this of course is that in many cases, the plists and the other XML should have slightly different indent rules. The standard plist formatting skips the first indent level.

Reply:
Agree, the opening `<plist>` tag should not be indented. I believe we are currently doing this correctly.